packages/components/src/Text/Text.tsx

```diff
@@ -1,28 +1,30 @@
-import React from 'react';
-import { Box, system } from '@marigold/system';
+import { createStyles } from '@marigold/system';
+import React, { HTMLAttributes } from 'react';
 
-type TextProps = {
+type TextProps = React.PropsWithChildren<{
+  as?: 'p' | 'span';
+  variant?: 'body' | 'heading';
   textColor?: string;
-};
+}>;
+
+type HtmlTextProps = Omit<HTMLAttributes<HTMLElement>, 'style'> & TextProps;
+
+const useStyles = createStyles('text');
 
-export const Text = system<TextProps, 'span'>(
-  ({
-    as = 'span',
-    variant = 'body',
-    textColor = 'inherit',
-    children,
-    ...props
-  }) => {
-    return (
-      <Box
-        {...props}
-        as={as}
-        themeSection="text"
-        variant={variant}
-        css={{ color: textColor }}
-      >
-        {children}
-      </Box>
-    );
-  }
-);
+export const Text: React.FC<HtmlTextProps> = ({
+  as = 'span',
+  variant = 'body',
+  textColor = 'inherit',
+  children,
+  ...props
+}: TextProps) => {
+  const classNames = useStyles({ variant, color: textColor });
+
+  const Cmp = as;
```

Comment:
This is fine for now, but we maybe have to add a helper or something to make this easier.